Todo.py

```diff
@@ -0,0 +1,203 @@
+
+    @todo.command()
+    async def add(self, ctx, *args):
+        '''Command to add a todo. Usage : <the thing todo>;<type of todo>;<assigned to / false>;<repost public channel / false>'''
+        with open(TODO_CHANNEL, "r") as file:
+            channelId = file.readlines()[0]
+            if channelId == None or channelId == "":
+                await ctx.channel.send("The todos' channel must be selected with the command " + PREFIX + "todo channel")
+                return
+        channel = bot.get_channel(int(channelId))
+
+        command = ""
+        for arg in args:
+            command += " {}".format(arg)
+        command = command.split(";")
+
+        todoType = None
+
+        with open(TODO_TYPES, "r") as file:
+            lines = file.readlines()
+            for line in lines:
+                line = line.split(';')
+                if command[1] == line[0]:
+                    todoType = line
+                    break
+        
+        if todoType != None:
+            embedColor = int(todoType[1], 16)
+        else:
+            embedColor = 0x28a745
+        
+        newEmbed = discord.Embed(description=command[0], url="", color=embedColor)
+        
+        command[2] = command[2].replace(' ', '') #TODO: Use dfind instead ?
+        if command[2] != "false":
+            if command[2].startswith("<@"):
+                user = ctx.guild.get_member(int(command[2][2:-1]))
+            else:
+                user = ctx.guild.get_member_named(command[2])
+            
+            if user != None:
+                newEmbed.add_field(name="Asssigned to", value=user.mention, inline=True)
+        
+        if command[3] != "false":
+            if command[3].startswith("<#"):
+                repostChannel = ctx.guild.get_channel(int(command[3][2:-1]))
+            else:
+                for chan in ctx.guild.channels:
+                    if(chan.name == command[3]):
+                        repostChannel = chan
+        else:
+            repostChannel = None
+        
+        newEmbed.set_footer(text=command[1])
+        if repostChannel != None:
+            publicTodo = await repostChannel.send(embed=newEmbed)
+            newEmbed.add_field(name="Public repost", value=repostChannel.mention + " : " + str(publicTodo.id), inline=True)
+        
+        message = await channel.send(embed=newEmbed)
+        await message.add_reaction(EMOJIS['wastebasket'])
+        await message.add_reaction(EMOJIS['check'])
+        await message.add_reaction(EMOJIS['hourglass'])
+        await ctx.message.delete()
+
+    @todo.command()
+    async def addtype(self, ctx, command):
+        '''Command to add a todo type.'''
+        command = command.split(";")
+
+        if command[1].startswith("#"):
+            command[1] = command[1][1:]
+        if len(command[1]) != 6:
+            if len(command[1]) != 3:
+                await ctx.send('The color must be in hexadecimal, like this **#ccc** or **#ff0000**')
```

Comment:
Should we add more "natural" names for the colors ? This could be useful not only for this project but for other commands too. May even save us, the coders some time ^^

Reply:
Hmm yeah not bad idea but, then how ? users will use text like "red" ? 
Also there is some basic colors already done see https://discordpy.readthedocs.io/en/latest/api.html#discord.Colour

Reply:
oh I didn't know. Well this may reveal itself useful one day